src/ClientMessage.elm

```diff
@@ -1,54 +1,54 @@
-module Message.ComponentMsg exposing (ComponentMsg(..), decoder, encode)
+module ClientMessage exposing (ClientMessage(..), decoder, encode)
 
-{-| The ComponentMsg module describes message formats used by components to communicate with the parent application.
+{-| The ClientMsg module describes message formats used by clients to communicate with the parent application.
 
-@docs ComponentMsg, decoder, encode
+@docs ClientMessage, decoder, encode
 
 -}
 
 import Dict exposing (Dict)
 import Json.Decode as Decode exposing (Decoder)
 import Json.Decode.Pipeline exposing (decode, required)
 import Json.Encode as Encode
-import Message.LabeledMsg as LabeledMsg
+import LabeledMessage
```

Comment:
As I was working on toast message, I added the type, encoders, and decoders in Message.ToastMsg.  It might be nice to have this root message module for code organization as the message types grow.  That said, I'm still an elm padwan and not sure if this is the best way to organize.

Reply:
Along the same lines, maybe it would be good to have top level modules of host, client, and message?

Again, take this with a grain of salt.

Reply:
For ToastMsg, since it's a client -> host message (ignore any action responses for now), I'd add all of the decoders and whatnot in this file directly, and add a variant to the `ClientMessage` type to capture it.

A good organizational strategy in Elm is to keep more code than you normally would in JavaScript in a single file and only split when things start to get awkward. Having a good type system makes putting more code in one place less of an anti-pattern. (there are limits, of course)